session cookies - undefined: err

I have been testing this and here are my findings. Base on `go get -u goa.design/goa/v3@cb1d437`

ran the following commands

`goa gen session_test/design`
`goa example session_test/design cmd`

compile
`go build ./cmd/user_mgmt`

generates error:

`gen/http/user_mgmt/server/encode_decode.go:307:6: undefined: err`

See full example `design.go` below
```go
package design

import (
	. "goa.design/goa/v3/dsl"
	_ "goa.design/plugins/v3/zerologger" // Enables the plugin
)

var _ = API("Account Management", func() {
	Title("User Account Service")
	Description("Creation, verification and login / logout for a user")
	Server("user_mgmt", func() {
		Host("localhost", func() {
			URI("http://localhost:8000")

		})
	})
})

var BasicAuth = BasicAuthSecurity("basic", func() {
	Description("Basic authentication used to authenticate security principal during sign in")
	// Scope("api:read", "Read-only access")
})

var NewUserPayload = Type("NewUserPayload", func() {
	Attribute("email", String, "a valid, working email address")
	Attribute("pwd1", String, "password")
	Attribute("pwd2", String, "password retyped")

	Required("email", "pwd1", "pwd2")
})

var _ = Service("UserMGMT", func() {
	Description("The user service allows the creation and mgmt of a user account.")

	Method("create", func() {
		Payload(NewUserPayload)
		Error("internal_server_error", func() { Description("there has been an internal error") })
		Error("email_exists", func() { Description("the email address is already in use.") })
		Error("passwords_do_not_match", func() { Description("the passwords entered do not match.") })

		Result(func() {
			Attribute("email", String, "email address that was used to sign up")
			Attribute("val_token_expire_at", String, "number of seconds until the validation token expires")
			Attribute("val_token_created_at", String, "the time at which the token was created")
			Required("email", "val_token_expire_at", "val_token_created_at")
		})

		HTTP(func() {
			POST("create")
			Response(StatusCreated, func() {
				//TODO Tag("outcome", "created") // CreateResult type to describe body
			})

			Response("internal_server_error", StatusInternalServerError)
			Response("email_exists", StatusBadRequest)
			Response("passwords_do_not_match", StatusBadRequest)
		})
	})

	Method("verify", func() {
		Error("BadRequest")

		Payload(func() {
			Attribute("token", String, "token that was received")
			Required("token")
		})

		HTTP(func() {
			GET("verify")
			Param("token")

			Response(StatusOK)
			Response("BadRequest", StatusBadRequest)
		})

	})

	Method("healthy", func() {
		Error("service_unavailable")
		HTTP(func() {
			GET("healthy")
			Response(StatusOK)
			Response("service_unavailable", StatusServiceUnavailable)
		})
	})

	Method("ready", func() {
		Error("service_unavailable")
		HTTP(func() {

			GET("ready")
			Response(StatusOK)
			Response("service_unavailable", StatusServiceUnavailable)

		})
	})

	Method("logout", func() {
		Payload(func() {
			Attribute("session_id", String, "Session ID")
			Required("session_id")
		})

		Result(func() {
			Attribute("session_id", String, "Session ID")
			Attribute("message", String, "User message")
			Required("session_id", "message")
		})

		HTTP(func() {
			POST("logout")
			Cookie("session_id:SID") // Load session ID from "SID" cookie

			Response(StatusOK, func() {
				//here to allow clearing of the session cookie
				Cookie("session_id:SID")
				CookieHTTPOnly()
				//CookieSecure()
				CookieMaxAge(-1)
			})
		})

	})

	Method("login", func() {
		Error("unauthorized", String, "Credentials are invalid")

		Security(BasicAuth)

		Payload(func() {
			//session_id here is to allow the login function to check if there is actually a token, then we compare to the token stored in the backend
			//if session_id is empty, then we login them in, setting the session token in the Result that is set below
			Attribute("session_id", String, "Session ID")

			UsernameField(1, "email", String, "email address that was used to sign up for an account", func() {
				Example("joe@domain.com")
				//Format(FormatEmail)
			})

			PasswordField(2, "password", String, "password you used to sign up for an account", func() {
				Example("password")
			})

			Required("email", "password")

		})

		//for setting generated session token if this is a new loging, store in the backend
		Result(func() {
			Attribute("session_id", String, "Session ID")
			Attribute("message", String, "User message")
			Required("session_id", "message")
		})

		HTTP(func() {

			POST("login")

			//here to allow reading of the session token if present in the request
			Cookie("session_id:SID")

			Response(StatusCreated, func() {
				//here to allow setting of the session token
				Cookie("session_id:SID")
				CookieHTTPOnly()
				//CookieSecure()
				CookieMaxAge(3600)
			})

			Response("unauthorized", StatusUnauthorized)
		})
	})

	Method("profile", func() {
		Payload(func() {
			Attribute("session_id", String, "Session ID")
			Required("session_id")
		})

		Result(func() {
			Attribute("email", String, "email")
			Attribute("profile_age", String, "profile age")
			Required("email", "profile_age")
		})

		HTTP(func() {
			GET("profile")
			Cookie("session_id:SID") // Load session ID from "SID" cookie
			Response(StatusOK)
		})

	})

	Files("/openapi.json", "./gen/http/openapi.json")
})

```

--- a/dsl/http.go
+++ b/dsl/http.go
@@ -461,9 +461,8 @@ func Cookie(name string, args ...interface{}) {
 //            Result(Account)
 //            HTTP(func() {
 //                Response(StatusCreated, func() {
-//                    Cookie("session:SID", String, func() {
-//                        CookieMaxAge(3600)
-//                    })
+//                    Cookie("session:SID", String)
+//                    CookieMaxAge(3600)
 //                })
 //            })
 //        })
@@ -491,9 +490,8 @@ func CookieMaxAge(n int) {
 //            Result(Account)
 //            HTTP(func() {
 //                Response(StatusCreated, func() {
-//                    Cookie("session:SID", String, func() {
-//                        CookieDomain("goa.design")
-//                    })
+//                    Cookie("session:SID", String)
+//                    CookieDomain("goa.design")
 //                })
 //            })
 //        })
@@ -521,9 +519,8 @@ func CookieDomain(d string) {
 //            Result(Account)
 //            HTTP(func() {
 //                Response(StatusCreated, func() {
-//                    Cookie("session:SID", String, func() {
-//                        CookiePath("/session")
-//                    })
+//                    Cookie("session:SID", String)
+//                    CookiePath("/session")
 //                })
 //            })
 //        })
@@ -550,9 +547,8 @@ func CookiePath(p string) {
 //            Result(Account)
 //            HTTP(func() {
 //                Response(StatusCreated, func() {
-//                    Cookie("session:SID", String, func() {
-//                        CookieSecure()
-//                    })
+//                    Cookie("session:SID", String)
+//                    CookieSecure()
 //                })
 //            })
 //        })
@@ -579,9 +575,8 @@ func CookieSecure() {
 //            Result(Account)
 //            HTTP(func() {
 //                Response(StatusCreated, func() {
-//                    Cookie("session:SID", String, func() {
-//                        CookieHTTPOnly()
-//                    })
+//                    Cookie("session:SID", String)
+//                    CookieHTTPOnly()
 //                })
 //            })
 //        })

--- a/http/codegen/server_decode_test.go
+++ b/http/codegen/server_decode_test.go
@@ -123,6 +123,16 @@ func TestDecode(t *testing.T) {
 		{"header-string-default-validate", testdata.PayloadHeaderStringDefaultValidateDSL, testdata.PayloadHeaderStringDefaultValidateDecodeCode},
 		{"header-primitive-string-default", testdata.PayloadHeaderPrimitiveStringDefaultDSL, testdata.PayloadHeaderPrimitiveStringDefaultDecodeCode},
 
+		{"cookie-string", testdata.PayloadCookieStringDSL, testdata.PayloadCookieStringDecodeCode},
+		{"cookie-string-validate", testdata.PayloadCookieStringValidateDSL, testdata.PayloadCookieStringValidateDecodeCode},
+
+		{"cookie-primitive-string-validate", testdata.PayloadCookiePrimitiveStringValidateDSL, testdata.PayloadCookiePrimitiveStringValidateDecodeCode},
+		{"cookie-primitive-bool-validate", testdata.PayloadCookiePrimitiveBoolValidateDSL, testdata.PayloadCookiePrimitiveBoolValidateDecodeCode},
+
+		{"cookie-string-default", testdata.PayloadCookieStringDefaultDSL, testdata.PayloadCookieStringDefaultDecodeCode},
+		{"cookie-string-default-validate", testdata.PayloadCookieStringDefaultValidateDSL, testdata.PayloadCookieStringDefaultValidateDecodeCode},
+		{"cookie-primitive-string-default", testdata.PayloadCookiePrimitiveStringDefaultDSL, testdata.PayloadCookiePrimitiveStringDefaultDecodeCode},
+
 		{"body-string", testdata.PayloadBodyStringDSL, testdata.PayloadBodyStringDecodeCode},
 		{"body-string-validate", testdata.PayloadBodyStringValidateDSL, testdata.PayloadBodyStringValidateDecodeCode},
 		{"body-user", testdata.PayloadBodyUserDSL, testdata.PayloadBodyUserDecodeCode},

--- a/http/codegen/service_data.go
+++ b/http/codegen/service_data.go
@@ -990,10 +990,13 @@ func buildPayloadData(e *expr.HTTPEndpointExpr, sd *ServiceData) *PayloadData {
 				sd.ServerTypeNames[serverBodyData.Name] = false
 				sd.ClientTypeNames[serverBodyData.Name] = false
 			}
-			for _, p := range paramsData {
-				if p.Validate != "" || needConversion(p.Type) {
-					mustValidate = true
-					break
+			mustValidate = len(cookiesData) > 0
+			if !mustValidate {
+				for _, p := range paramsData {
+					if p.Validate != "" || needConversion(p.Type) {
+						mustValidate = true
+						break
+					}
 				}
 			}
 			if !mustValidate {
@@ -1007,14 +1010,6 @@ func buildPayloadData(e *expr.HTTPEndpointExpr, sd *ServiceData) *PayloadData {
 			if !mustValidate {
 				for _, h := range headersData {
 					if h.Validate != "" || h.Required || needConversion(h.Type) {
-						mustValidate = true
-						break
-					}
-				}
-			}
-			if !mustValidate {
-				for _, c := range cookiesData {
-					if c.Validate != "" || c.Required || needConversion(c.Type) {
 						mustValidate = true
 						break
 					}

--- a/http/codegen/testdata/payload_decode_functions.go
+++ b/http/codegen/testdata/payload_decode_functions.go
@@ -3411,6 +3411,217 @@ func DecodeMethodHeaderPrimitiveStringDefaultRequest(mux goahttp.Muxer, decoder 
 }
 `
 
+var PayloadCookieStringDecodeCode = `// DecodeMethodCookieStringRequest returns a decoder for requests sent to the
+// ServiceCookieString MethodCookieString endpoint.
+func DecodeMethodCookieStringRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {
+	return func(r *http.Request) (interface{}, error) {
+		var (
+			c2  *string
+			err error
+			c   *http.Cookie
+		)
+		c, err = r.Cookie("c")
+		var c2Raw string
+		if c != nil {
+			c2Raw = c.Value
+		}
+		if c2Raw != "" {
+			c2 = &c2Raw
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload := NewMethodCookieStringPayload(c2)
+
+		return payload, nil
+	}
+}
+`
+
+var PayloadCookieStringValidateDecodeCode = `// DecodeMethodCookieStringValidateRequest returns a decoder for requests sent
+// to the ServiceCookieStringValidate MethodCookieStringValidate endpoint.
+func DecodeMethodCookieStringValidateRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {
+	return func(r *http.Request) (interface{}, error) {
+		var (
+			c2  *string
+			err error
+			c   *http.Cookie
+		)
+		c, err = r.Cookie("c")
+		var c2Raw string
+		if c != nil {
+			c2Raw = c.Value
+		}
+		if c2Raw != "" {
+			c2 = &c2Raw
+		}
+		if c2 != nil {
+			err = goa.MergeErrors(err, goa.ValidatePattern("c2", *c2, "cookie"))
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload := NewMethodCookieStringValidatePayload(c2)
+
+		return payload, nil
+	}
+}
+`
+
+var PayloadCookiePrimitiveStringValidateDecodeCode = `// DecodeMethodCookiePrimitiveStringValidateRequest returns a decoder for
+// requests sent to the ServiceCookiePrimitiveStringValidate
+// MethodCookiePrimitiveStringValidate endpoint.
+func DecodeMethodCookiePrimitiveStringValidateRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {
+	return func(r *http.Request) (interface{}, error) {
+		var (
+			c2  string
+			err error
+			c   *http.Cookie
+		)
+		c, err = r.Cookie("c")
+		if err == http.ErrNoCookie {
+			err = goa.MergeErrors(err, goa.MissingFieldError("c", "cookie"))
+		} else {
+			c2 = c.Value
+		}
+		if !(c2 == "val") {
+			err = goa.MergeErrors(err, goa.InvalidEnumValueError("c2", c2, []interface{}{"val"}))
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload := c
+
+		return payload, nil
+	}
+}
+`
+
+var PayloadCookiePrimitiveBoolValidateDecodeCode = `// DecodeMethodCookiePrimitiveBoolValidateRequest returns a decoder for
+// requests sent to the ServiceCookiePrimitiveBoolValidate
+// MethodCookiePrimitiveBoolValidate endpoint.
+func DecodeMethodCookiePrimitiveBoolValidateRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {
+	return func(r *http.Request) (interface{}, error) {
+		var (
+			c2  bool
+			err error
+			c   *http.Cookie
+		)
+		c, err = r.Cookie("c")
+		{
+			var c2Raw string
+			if c != nil {
+				c2Raw = c.Value
+			}
+			if c2Raw == "" {
+				err = goa.MergeErrors(err, goa.MissingFieldError("c", "cookie"))
+			}
+			v, err2 := strconv.ParseBool(c2Raw)
+			if err2 != nil {
+				err = goa.MergeErrors(err, goa.InvalidFieldTypeError("c2", c2Raw, "boolean"))
+			}
+			c2 = v
+		}
+		if !(c2 == true) {
+			err = goa.MergeErrors(err, goa.InvalidEnumValueError("c2", c2, []interface{}{true}))
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload := c
+
+		return payload, nil
+	}
+}
+`
+
+var PayloadCookieStringDefaultDecodeCode = `// DecodeMethodCookieStringDefaultRequest returns a decoder for requests sent
+// to the ServiceCookieStringDefault MethodCookieStringDefault endpoint.
+func DecodeMethodCookieStringDefaultRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {
+	return func(r *http.Request) (interface{}, error) {
+		var (
+			c2  string
+			err error
+			c   *http.Cookie
+		)
+		c, err = r.Cookie("c")
+		var c2Raw string
+		if c != nil {
+			c2Raw = c.Value
+		}
+		if c2Raw != "" {
+			c2 = c2Raw
+		} else {
+			c2 = "def"
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload := NewMethodCookieStringDefaultPayload(c2)
+
+		return payload, nil
+	}
+}
+`
+
+var PayloadCookieStringDefaultValidateDecodeCode = `// DecodeMethodCookieStringDefaultValidateRequest returns a decoder for
+// requests sent to the ServiceCookieStringDefaultValidate
+// MethodCookieStringDefaultValidate endpoint.
+func DecodeMethodCookieStringDefaultValidateRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {
+	return func(r *http.Request) (interface{}, error) {
+		var (
+			c2  string
+			err error
+			c   *http.Cookie
+		)
+		c, err = r.Cookie("c")
+		var c2Raw string
+		if c != nil {
+			c2Raw = c.Value
+		}
+		if c2Raw != "" {
+			c2 = c2Raw
+		} else {
+			c2 = "def"
+		}
+		if !(c2 == "def") {
+			err = goa.MergeErrors(err, goa.InvalidEnumValueError("c2", c2, []interface{}{"def"}))
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload := NewMethodCookieStringDefaultValidatePayload(c2)
+
+		return payload, nil
+	}
+}
+`
+var PayloadCookiePrimitiveStringDefaultDecodeCode = `// DecodeMethodCookiePrimitiveStringDefaultRequest returns a decoder for
+// requests sent to the ServiceCookiePrimitiveStringDefault
+// MethodCookiePrimitiveStringDefault endpoint.
+func DecodeMethodCookiePrimitiveStringDefaultRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {
+	return func(r *http.Request) (interface{}, error) {
+		var (
+			c2  string
+			err error
+			c   *http.Cookie
+		)
+		c, err = r.Cookie("c")
+		if err == http.ErrNoCookie {
+			err = goa.MergeErrors(err, goa.MissingFieldError("c", "cookie"))
+		} else {
+			c2 = c.Value
+		}
+		if err != nil {
+			return nil, err
+		}
+		payload := c
+
+		return payload, nil
+	}
+}
+`
+
 var PayloadBodyStringDecodeCode = `// DecodeMethodBodyStringRequest returns a decoder for requests sent to the
 // ServiceBodyString MethodBodyString endpoint.
 func DecodeMethodBodyStringRequest(mux goahttp.Muxer, decoder func(*http.Request) goahttp.Decoder) func(*http.Request) (interface{}, error) {

--- a/http/codegen/testdata/payload_dsls.go
+++ b/http/codegen/testdata/payload_dsls.go
@@ -1754,6 +1754,111 @@ var PayloadHeaderPrimitiveStringDefaultDSL = func() {
 	})
 }
 
+var PayloadCookieStringDSL = func() {
+	Service("ServiceCookieString", func() {
+		Method("MethodCookieString", func() {
+			Payload(func() {
+				Attribute("c", String)
+			})
+			HTTP(func() {
+				GET("/")
+				Cookie("c")
+			})
+		})
+	})
+}
+
+var PayloadCookieStringValidateDSL = func() {
+	Service("ServiceCookieStringValidate", func() {
+		Method("MethodCookieStringValidate", func() {
+			Payload(func() {
+				Attribute("c", String, func() {
+					Pattern("cookie")
+				})
+			})
+			HTTP(func() {
+				GET("/")
+				Cookie("c")
+			})
+		})
+	})
+}
+
+var PayloadCookiePrimitiveStringValidateDSL = func() {
+	Service("ServiceCookiePrimitiveStringValidate", func() {
+		Method("MethodCookiePrimitiveStringValidate", func() {
+			Payload(String, func() {
+				Enum("val")
+			})
+			HTTP(func() {
+				GET("/")
+				Cookie("c")
+			})
+		})
+	})
+}
+
+var PayloadCookiePrimitiveBoolValidateDSL = func() {
+	Service("ServiceCookiePrimitiveBoolValidate", func() {
+		Method("MethodCookiePrimitiveBoolValidate", func() {
+			Payload(Boolean, func() {
+				Enum(true)
+			})
+			HTTP(func() {
+				GET("/")
+				Cookie("c")
+			})
+		})
+	})
+}
+
+var PayloadCookieStringDefaultDSL = func() {
+	Service("ServiceCookieStringDefault", func() {
+		Method("MethodCookieStringDefault", func() {
+			Payload(func() {
+				Attribute("c", String, func() {
+					Default("def")
+				})
+			})
+			HTTP(func() {
+				GET("/")
+				Cookie("c")
+			})
+		})
+	})
+}
+
+var PayloadCookieStringDefaultValidateDSL = func() {
+	Service("ServiceCookieStringDefaultValidate", func() {
+		Method("MethodCookieStringDefaultValidate", func() {
+			Payload(func() {
+				Attribute("c", String, func() {
+					Default("def")
+					Enum("def")
+				})
+			})
+			HTTP(func() {
+				GET("/")
+				Cookie("c")
+			})
+		})
+	})
+}
+
+var PayloadCookiePrimitiveStringDefaultDSL = func() {
+	Service("ServiceCookiePrimitiveStringDefault", func() {
+		Method("MethodCookiePrimitiveStringDefault", func() {
+			Payload(String, func() {
+				Default("def")
+			})
+			HTTP(func() {
+				GET("")
+				Cookie("c")
+			})
+		})
+	})
+}
+
 var PayloadJWTAuthorizationHeaderDSL = func() {
 	var JWT = JWTSecurity("jwt", func() {
 		Scope("api:read")

